src/main/java/io/kestra/plugin/fs/tcp/Trigger.java

```diff
@@ -0,0 +1,148 @@
+import reactor.core.publisher.Flux;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+
+@Setter
+@SuperBuilder
+@Getter
+@ToString
+@EqualsAndHashCode(callSuper = true)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Listen for TCP messages on port 5000",
+            code = """
+                id: tcp_trigger_flow
+                namespace: company.team
+
+                tasks:
+                  - id: log
+                    type: io.kestra.plugin.core.debug.Return
+                    format: "Received TCP: {{ trigger.message }}"
+
+                triggers:
+                  - id: tcp
+                    type: io.kestra.plugin.fs.tcp.TcpRealtimeTrigger
+                    port: 5000
+                """
+        )
+    }
+)
+public class Trigger extends RealtimeTrigger<Trigger.Output> {
+    @PluginProperty
+    private Integer port;
+
+    public Trigger() {
+        super();
+    }
+
+    @Override
+    protected void listen(RunContext runContext,
+                          TriggerContext triggerContext,
+                          ConditionContext conditionContext) {
+        runContext.logger().info("TCP RealtimeTrigger listening on port {}", port);
+
+        try (ServerSocket serverSocket = new ServerSocket(port)) {
+            while (!Thread.currentThread().isInterrupted()) {
+                Socket client = serverSocket.accept();
+                runContext.logger().debug("New TCP client: {}", client.getRemoteSocketAddress());
+
+                // spawn a client handler
+                new Thread(() -> handleClient(runContext, triggerContext, conditionContext, client)).start();
+            }
+        } catch (Exception e) {
+            runContext.logger().error("TCP Trigger error", e);
+        }
+    }
+
+    void handleClient(RunContext runContext,
+                      TriggerContext triggerContext,
+                      ConditionContext conditionContext,
+                      Socket client) {
+        try (BufferedReader reader = new BufferedReader(
+            new InputStreamReader(client.getInputStream(), StandardCharsets.UTF_8))) {
+
+            StringBuilder message = new StringBuilder();
+            String line;
+            while ((line = reader.readLine()) != null) {
+                message.append(line).append("\n");
+            }
+
+            emitEvent(runContext, conditionContext, triggerContext, Output.builder()
+                .port(port)
+                .message(message.toString().trim())
+                .remoteAddress(client.getInetAddress().getHostAddress())
+                .build());
+
+        } catch (Exception e) {
+            runContext.logger().error("Error handling TCP client", e);
+        }
+    }
+
+    @Override
+    public Publisher<Execution> evaluate(ConditionContext conditionContext, TriggerContext context) throws Exception {
```

Comment:
The `evaluate()` method duplicates the socket listening logic from `listen()`.
Consider reusing the same implementation or extracting a shared helper.
The Debezium plugin's realtime trigger wraps its core logic with a reactive Flux
rather than maintaining two separate implementations.